### PR TITLE
[SPARK-26246][SQL][FOLLOWUP] Inferring TimestampType from JSON

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

Added new JSON option `inferTimestamp` (`true` by default) to control inferring of `TimestampType` from string values.

## How was this patch tested?

Add new UT to `JsonInferSchemaSuite`.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala [hunk 1]
```diff
@@ -117,6 +117,12 @@ private[sql] class JSONOptions(
    */
   val pretty: Boolean = parameters.get("pretty").map(_.toBoolean).getOrElse(false)
 
+  /**
+   * Enables inferring of TimestampType from strings matched to the timestamp pattern
+   * defined by the timestampFormat option.
+   */
+  val inferTimestamp: Boolean = parameters.get("inferTimestamp").map(_.toBoolean).getOrElse(true)
```
Comment (xy2953396112) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala]: rename inferTimestamp  for inferTimestampType ? 
```suggestion
  val inferTimestampType: Boolean = parameters.get("inferTimestamp").map(_.toBoolean).getOrElse(true)
```

File: docs/sql-migration-guide-upgrade.md [hunk 2]
```diff
@@ -40,6 +40,9 @@ displayTitle: Spark SQL Upgrading Guide
...
+
+  - Since Spark 3.0, JSON datasource and JSON function `schema_of_json` infer TimestampType from string values if they matches to the pattern defined by the JSON option `timestampFormat`. Set JSON option `inferTimestamp` to `false` to disable such type inferring.
```
Comment (HyukjinKwon) [inline docs/sql-migration-guide-upgrade.md]: typo: they matches -> they match

**Comment by HyukjinKwon** [comment]
Merged to master.

File: docs/sql-migration-guide-upgrade.md (hunk 2 above)
Comment (MaxGekk) [reply docs/sql-migration-guide-upgrade.md]: Will fix next time

**Pull request closed without merging**